Library/Homebrew/cmd/update-report.rb

```diff
@@ -6,26 +5,20 @@
 
 module Homebrew
   def update_report
-    # migrate to new directories based tap structure
-    migrate_taps
+    CoreTap.ensure_installed! :quiet => false
 
     hub = ReporterHub.new
 
-    begin
-      master_reporter = Reporter.new(CoreTap.instance)
-    rescue Reporter::ReporterRevisionUnsetError => e
-      raise e if ARGV.homebrew_developer?
+    initial_revision = ENV["HOMEBREW_UPDATE_BEFORE"].to_s
+    current_revision = ENV["HOMEBREW_UPDATE_AFTER"].to_s
+    if initial_revision.empty? || current_revision.empty?
       odie "update-report should not be called directly!"
     end
 
-    if master_reporter.updated?
-      initial_short = shorten_revision(master_reporter.initial_revision)
-      current_short = shorten_revision(master_reporter.current_revision)
-      puts "Updated Homebrew from #{initial_short} to #{current_short}."
-      hub.add(master_reporter)
+    if brew_updated = initial_revision != current_revision
```

Comment:
Until we have a `brew` repo separation the `brew` name here also isn't super clear to me.